Draft PR for configuring RTP extensions

juberti opened the pull request:
Fixes #1032 
Fixes #1031
@aboba @ekr @fippo PTAL

juberti committed:
Draft PR for configuring RTP extensions

fippo commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -3481,13 +3481,13 @@ as specified in <xref target="RFC5245" format="default"/>.</li>
 	      be used in outgoing RTP packets when sending each specified
 	      header extension. If any indicated RTP header extension is
-	      not supported by the local implementation, it <bcp14>MUST</bcp14> be
+	      not enabled in the local implementation, it <bcp14>MUST</bcp14> be
```
this one can remain "supported by" since it means "not implemented"

juberti replied:
yeah, I wondered about that, but shouldn't `enabled` also factor in here? That is, if somehow we get a SDP with a header extension that isn't enabled, shouldn't that header extension ID be ignored?

juberti commented:
There's been some discussion on the rtcweb list on this, I need to go check back in on that discussion.

henbos commented:
Friendly ping

fippo commented:
Ping. We have the [api-owner LGTM](https://groups.google.com/a/chromium.org/g/blink-dev/c/KRfPpAB56yk) to ship the extension in Chromium-based browsers now

juberti committed:
Update per Christer feedback

per
> b) Extend the PR below to handle all the cases where the application might want to override "supported" (FEC and RTCP are the only remaining ones, I believe). We could use "supported by the application" as Tim mentions to make the change more editorial.

juberti committed:
Update draft-ietf-rtcweb-jsep.xml

self review

juberti committed:
Update draft-ietf-rtcweb-jsep.xml

self review

juberti commented:
updated the PR per the email thread, PTAL

fippo commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -1236,8 +1236,8 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host ]]></sourcecode>
           contains descriptions of the media added to this
-          PeerConnection, the codec/RTP/RTCP options negotiated for
-          this session, and any candidates that have been gathered by
+          PeerConnection, the codec, RTP, and RTCP options supported by the application,
```
```suggestion
          PeerConnection, the codecs, RTP, and RTCP options supported by the application,
```

juberti replied:
see above

fippo commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -1182,11 +1182,14 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host ]]></sourcecode>
-          that have been gathered by the ICE agent. An options
+          contains an offer per <xref target="RFC3264" format="default"/> that reflects
+          the current configuration of the session, including descriptions of the
+          media added to this PeerConnection and the codec, RTP, and RTCP
```
```suggestion
          media added to this PeerConnection and the codecs, RTP, and RTCP
```

juberti replied:
here, "codec" is an adjective modifying "options"

henbos commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -2272,11 +2272,9 @@ as specified in <xref target="RFC5245" format="default"/>.</li>
             recent answer <bcp14>MUST</bcp14> be added after all existing formats.</li>
             <li>The RTP header extensions <bcp14>MUST</bcp14> only include those that
-            are present in the most recent answer.</li>
+            are supported by the application on the associated transceiver.</li>
```
I think this makes sense, assuming the W3C spec can say that calling the JS method `setHeaderExtensionsToNegotiate()` will "Set which RTP header extensions are _supported by the application on the associated transceiver_. This will affect which RTP header extensions are negotiated, as defined by [JSEP reference to here]."

aboba replied:
Yes, that is what we would say in the WebRTC-Extensions spec.

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -2604,15 +2602,15 @@ as specified in <xref target="RFC5245" format="default"/>.</li>
             decoded, an "a=imageattr" line, as specified in
             <xref target="sec.imageattr" format="default"/>.</li>
-            <li>For each supported RTP header extension that is present
+            <li>For each RTP header extension supported by the application and present
```
agreed, but let me try to find a cleaner way to write this

juberti committed:
Pare down

juberti committed:
fix errors

juberti committed:
fix date

juberti commented:
Shipped as part of https://www.ietf.org/archive/id/draft-uberti-rtcweb-rfc8829bis-05.txt
